src/sys/stat.rs

```diff
@@ -134,3 +150,99 @@ pub fn fstatat<P: ?Sized + NixPath>(dirfd: RawFd, pathname: &P, f: AtFlags) -> R
     Ok(dst)
 }
 
+/// change permissions of a file 
+/// ([see chmod(2)](http://man7.org/linux/man-pages/man2/chmod.2.html))
+pub fn chmod<P: ?Sized + NixPath>(pathname: &P, mode: Mode) -> Result<()> {
+    let res = try!(pathname.with_nix_path(|cstr| {
+        unsafe { libc::chmod(cstr.as_ptr(), mode.bits()) }
+    }));
+
+    Errno::result(res).map(drop)
+}
+
+/// change permissions of a file 
+/// ([see fchmod(2)](http://man7.org/linux/man-pages/man2/fchmod.2.html))
+pub fn fchmod(fd: RawFd, mode: Mode) -> Result<()> {
+    let res = unsafe { libc::fchmod(fd, mode.bits()) };
+
+    Errno::result(res).map(drop)
+}
+
+/// change permissions of a file 
+/// ([see fchmodat(2)](http://man7.org/linux/man-pages/man2/fchmodat.2.html))
+pub fn fchmodat<P: ?Sized + NixPath>(dirfd: RawFd, pathname: &P, mode: Mode, flags: AtFlags) -> Result<()> {
+    let res = try!(pathname.with_nix_path(|cstr| {
+        unsafe {
+            libc::fchmodat(dirfd,
+                           cstr.as_ptr(),
+                           mode.bits(),
+                           flags.bits())
+        }
+    }));
+
+    Errno::result(res).map(drop)
+}
+
+#[cfg(target_os = "linux")]
+mod linux {
+    use {Errno, Result, NixPath};
+    use std::os::unix::io::RawFd;
+    use libc;
+    use fcntl::AtFlags;
+    use sys::time::TimeSpec;
+
+    pub enum UtimeSpec {
+        Now,
+        Omit,
+        Time(TimeSpec)
+    }
+
+    fn to_timespec(time: &UtimeSpec) -> libc::timespec {
+        match time {
+            &UtimeSpec::Now => libc::timespec {
+                tv_sec: 0,
+                tv_nsec: libc::UTIME_NOW,
+            },
+            &UtimeSpec::Omit => libc::timespec {
+                tv_sec: 0,
+                tv_nsec: libc::UTIME_OMIT,
+            },
+            &UtimeSpec::Time(spec) => *spec.as_ref()
+        }
+    }
+
+    /// Change file timestamps with nanosecond precision
+    /// (see [utimensat(2)](http://man7.org/linux/man-pages/man2/utimensat.2.html))
+    pub fn utimensat<P: ?Sized + NixPath>(dirfd: RawFd,
```

Comment:
`dirfd` and `pathname` actually combo together to form a ternary input type: one allowing a `RawFd`, another a relative path from the current working directory, and a third using an absolute path. I think we should use a ternary enum type here instead of the `dirfd` and `pathname` inputs to make this function call safer.